### PR TITLE
fix(core): type incorrect assumption of input always existing

### LINKED ISSUE #12001: Console errors throwing in fd-multi-input
#### Is this a bug, enhancement, or feature request?
It is a bug

Component affected: https://sap.github.io/fundamental-ngx/#/core/multi-input

#### Describe your proposal.

Console errors throwing in production in this function 
xe._makeSearchTermChangeAnnouncements
with error as Cannot read properties of undefined (reading 'length') 

#### Can you handle that on the application side
No

#### Which versions of Angular and Fundamental Library for Angular are affected? Please, specify the exact version. (If this is a feature request, use current version.)

Angular Version: 15.2.9
ngx version: 0.43.21

#### Is there anything else we should know?

**_IMPORTANT: Please refrain from providing links or screenshots of SAP's internal information, as this project is open-source, and its contents are accessible to anyone._**

### PR DESCRIPTION
**Pull request by ZsDeak-SAP**
## Related Issue(s)
closes #12001 

## Description

Input might be undefined, we can't assume it's not. Added an example, if you type `undefined` to that input, on `main` it'll reproduce the error of the linked issue. \
Type of filterFn is wrong on multi-input.component (this['dropdownValues']) as it contains a package private type. This is why in the example I used anys

### CODE REVIEW

**Comment by alexhristov14** [comment]
> @droshev [this change](https://github.com/SAP/fundamental-ngx/pull/12217) i suspect failed some e2e tests, so i tried to fix them as well

I've created a new PR for the e2e testing commit #12253 so you can revert it from this one and only keep the changes for the multi-input issue

File: libs/docs/core/multi-input/examples/multi-input-example/multi-input-example.component.ts [hunk 1]
```diff
@@ -32,4 +33,8 @@ export class MultiInputExampleComponent {
     searchTermChange(text: string): void {
         console.log(text);
     }
+
+    customFilter(currentArray: (string | any)[], searchTerm: string): any {
```
Comment (dpavlenishvili) [inline libs/docs/core/multi-input/examples/multi-input-example/multi-input-example.component.ts:37]: update return type

Comment (ZsDeak-SAP) [reply libs/docs/core/multi-input/examples/multi-input-example/multi-input-example.component.ts:37]: > Type of filterFn is wrong on multi-input.component (this['dropdownValues']) as it contains a package private type. This is why in the example I used anys

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
The linked issue and the comment thread on that issue, and the comment thread here, don't make it clear exactly what this PR is trying to solve. I think a stackblitz demonstrating the problem in the linked issue would go a long way. But until we have a clear explanation, I don't think we can move forward with merging this

**Comment by ZsDeak-SAP** [comment]
@mikerodonnell89 I hope you see how it's difficult to create a stackblitz to an wrongly working internal function. nevertheless here you can [find an example](https://stackblitz.com/edit/zdi7tq?file=src%2Fapp%2Fmulti-input-example.component.ts). type in 'undefined' and you'll get a console error. This example is partly incorrect, as `filterFn` would expect an array returning function, and does have a default value, but only through that can I reach (right now) the `multiAnnouncerOptions` input. 

`multiAnnouncerOptions` does not have a default value and is not a required input, and as such we cannot assume it to exist. this will cause the console error above.

File: libs/docs/core/multi-input/examples/multi-input-example/multi-input-example.component.ts (hunk 1 above)
Comment (mikerodonnell89) [reply libs/docs/core/multi-input/examples/multi-input-example/multi-input-example.component.ts:37]: I don't think we want to include an example of a deliberately broken filter function in our docs

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
Looks good but I think let's remove the new example you added. There's already the custom filter example, and I think the `return searchTerm === 'undefined' ? undefined` is going to be confusing to anyone who doesn't know the (very specific) context of this issue

**Comment by droshev** [comment]
@ZsDeak-SAP did you see what @mikerodonnell89 suggested?

**Comment by ZsDeak-SAP** [comment]
sorry, yes, fixed

**Comment by droshev** [comment]
@mikerodonnell89 Can you review the latest changes?

**Pull request merged**